### PR TITLE
Implementation to be updated in ApplicationPattern for the domain-name

### LINKED ISSUE #350: Implementation to be updated in ApplicationPattern for the domain-name
Require changes in the following classes to support the domain-name attribute , 

1. In the applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js , 
      - [ ] domain-name attribute needs to be included 
      - [ ] createTcpClientInterfaceAsync, setRemoteAddressAsync function logic needs to include 
            -  domain-name(and exclude ipv-4-address) if the incoming request consists of a domain-name
            -  ipv-4-address(and exclude domain-name) if the incoming request consists of a ipv-4-address
      - [ ] getRemoteAddressAsync function logic needs to be updated to look into one step deep in the “remote-address” field to figure the configured address.

2. In the applicationPattern/onfModel/models/layerProtocols/TcpServerInterface.js, 
      - [ ] getLocalAddressAsync function logic needs to be updated to look into one step deep in the “local-address” field to figure the configured address.
      - [ ] Include a new function setLocalAddressAsync that configures the local-address(either ipv-4-address or domain-name). This will be helpful in configuring the own address from the “register-yourself” operation

### PR DESCRIPTION
**Pull request by vanithavalluripalli9**
Fixes #350

### CODE REVIEW

File: server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/OperationClientInterface.js [hunk 1]
```diff
@@ -330,8 +331,29 @@ class OperationClientInterface extends layerProtocol {
...
-
+/**
+ * @description This function returns the remote address configured .
+ * @param {String} remoteAddress : remote address of the tcp client .
+ * @returns {promise} string {remoteAddress}
+ **/
+function getConfiguredRemoteAddress(remoteAddress) {
+    return new Promise(async function (resolve, reject) {
+        try {
+            if ("domain-name" in remoteAddress) {
```
Comment (PrathibaJee) [inline server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/OperationClientInterface.js:344]: Please remove the hardcoded values. Instead , declare them as a constant in the module applicationPattern\onfModel\constants\OnfAttributes.js and utilize the declared constant. In this way , we can utilize the attribute name globally across many modules.

Comment (PrathibaJee) [inline server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/OperationClientInterface.js:344]: Please remove the hardcoded values. Instead , declare them as a constant in the module applicationPattern\onfModel\constants\OnfAttributes.js and utilize the declared constant. In this way , we can utilize the attribute name globally across many modules.

File: server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js [hunk 2]
```diff
@@ -236,5 +220,44 @@ class TcpClientInterface extends layerProtocol {
...
+ * @param {String} remoteAddress : remote address of the tcp client .
+ * @returns {promise} list {paths}
+ **/
+function getPaths(tcpClientUuid, remoteAddress, addressToBeDeleted) {
+    return new Promise(async function (resolve, reject) {
+        let paths = [];
+        let remoteAddressPath;
+        let pathOfAddressToBeDeleted;
+        try {
+            if ("domain-name" in remoteAddress) {
```
Comment (PrathibaJee) [inline server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js:242]: Please remove the hardcoded values. Instead , declare them as a constant in the module applicationPattern\onfModel\constants\OnfAttributes.js and utilize the declared constant. In this way , we can utilize the attribute name globally across many modules.

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
Apart from two minor review comments no further comments. Please fix the mentioned changes. Many thanks.

File: server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/OperationClientInterface.js (hunk 1 above)
Comment (vanithavalluripalli9) [reply server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/OperationClientInterface.js]: Thank you for the comment.

Made changes as suggested , removed the hard coded values.

File: server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js (hunk 2 above)
Comment (vanithavalluripalli9) [reply server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js]: Thank you for the comment.

Made changes as suggested , removed the hard coded values.

**Review by PrathibaJee — APPROVED** [review]

**Pull request merged**